file-formats/gsmp/type/zif_aff_gsmp_v1.intf.abap

```diff
@@ -0,0 +1,153 @@
+    ty_provider_type  TYPE c LENGTH 4,
+
+    "! <p class="shorttext">Execution Mode</p>
+    "! Execution mode
+    "! $values {@link zif_aff_gsmp_v1.data:co_execution_mode}
+    ty_execution_mode TYPE c LENGTH 1,
+
+
+    "! <p class="shorttext">Provider Position</p>
+    "! Provider position
+    "! $values {@link zif_aff_gsmp_v1.data:co_provider_pos}
+    ty_provider_pos   TYPE i.
+
+
+  CONSTANTS:
+
+    "! <p class="shorttext">Provider Type</p>
+    "! Provider type
+    BEGIN OF co_provider_type,
+
+      "! <p class="shorttext">Class</p>
+      "! Class
+      "! $enumValue 'CLAS'
+      type_class    TYPE ty_provider_type VALUE 'CLAS',
+
+      "! <p class="shorttext">CDS View</p>
+      "! CDS view
+      "! $enumValue 'CDSA'
+      type_cds_view TYPE ty_provider_type VALUE 'CDSA',
+
+    END OF co_provider_type,
+
+
+
+    "! <p class="shorttext">Execution Mode</p>
+    "! Execution mode
+    BEGIN OF co_execution_mode,
+
+      "! <p class="shorttext">System</p>
+      "! System
+      "! $enumValue 'S'
+      system      TYPE ty_execution_mode VALUE 'S',
+
+      "! <p class="shorttext">Instance</p>
+      "! Instance
+      "! $enumValue 'I'
+      instance    TYPE ty_execution_mode VALUE 'I',
+
+      "! <p class="shorttext">Application</p>
+      "! Application
+      "! $enumValue 'A'
+      application TYPE ty_execution_mode VALUE 'A',
+
+      "! <p class="shorttext">None</p>
+      "! No execution. Only model. Data from external.
+      "! $enumValue 'N'
+      none        TYPE ty_execution_mode VALUE 'N',
+
+    END OF co_execution_mode,
+
+
+
+
+    "! <p class="shorttext">Provider Position</p>
+    "! Provider position
+    BEGIN OF co_provider_pos,
+
+      "! <p class="shorttext">Availability</p>
+      "! Availability
+      "! $enumValue '1'
+      availability TYPE ty_provider_pos VALUE 1,
+
+      "! <p class="shorttext">High</p>
+      "! High
+      "! $enumValue '200'
+      high         TYPE ty_provider_pos VALUE 200,
+
+      "! <p class="shorttext">Normal</p>
+      "! Normal
+      "! $enumValue '300'
+      normal       TYPE ty_provider_pos VALUE 300,
+
+
+      "! <p class="shorttext">Lown</p>
+      "! Low
+      "! $enumValue '400'
+      low          TYPE ty_provider_pos VALUE 400,
+
+    END OF co_provider_pos.
+
+
+  TYPES:
+
+    "! <p class="shorttext">Metric Provider</p>
+    "! Metric Provider (GSMP) v1
+    BEGIN OF ty_main,
+
+      "! $required
+      format_version TYPE zif_aff_types_v1=>ty_format_version,
+
+
+      "! <p class="shorttext">Header</p>
+      "! Header
+      "! $required
+      header         TYPE zif_aff_types_v1=>ty_header_60_cloud,
+
+
+      "! <p class="shorttext">Provider Type</p>
+      "! Provider type
+      "! $required
+      provider_type  TYPE ty_provider_type,
+
+
+      "! <p class="shorttext">Implementation</p>
+      "! Implementation
+      "! $required
+      implementation TYPE zif_aff_types_v1=>ty_object_name_30,
```

Comment:
Which component do you mean with 'name'? Generally, I prefer a substructure instead of having many single components. But in this case I struggle a bit with the name 'provider' since the whole object is a (Metric) "provider"

Reply:
If provider type = CLAS then field implementation hold a class name. If provider_type = CDSA field implementation hold the name of cds view. SO it would make sense to tie these field together in a structure